go/vt/topo/topoproto/tablet.go

```diff
@@ -227,6 +227,25 @@ func MakeStringTypeList(types []topodatapb.TabletType) []string {
 	return strs
 }
 
+// MakeUniqueStringTypeList returns a unique list of strings that match
+// the input list -- with duplicate types removed.
+// This is needed as some types are aliases for others, like BATCH and
+// RDONLY, so e.g. rdonly shows up twice in the list when using
+// AllTabletTypes.
+func MakeUniqueStringTypeList(types []topodatapb.TabletType) []string {
+	strs := make([]string, 0, len(types))
+	seen := make(map[string]bool)
```

Comment:
if we end up keeping this, `map[string]struct{}` is better (empty struct is zero bytes whereas bool is 1)